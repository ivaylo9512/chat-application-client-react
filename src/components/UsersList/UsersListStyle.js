import styled from 'styled-components'

export const Container = styled.div`
    width: 100%;
    flex: 1;
    display: flex;
    padding-bottom: 3vh;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    box-sizing: border-box;
    
    span{
        font-size: 1.1vw;
    }
`