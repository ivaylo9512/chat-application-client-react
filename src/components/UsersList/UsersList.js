import User from 'components/User/User'
import { useSelector } from 'react-redux';
import { getCurrentUsers, getCurrentPage } from 'app/slices/usersSlice';
import { Container } from './UsersListStyle';

const UsersList = () => {
    const users = useSelector(getCurrentUsers);
    const page = useSelector(getCurrentPage);

    return (
        <Container>
            {users && (users.length == 0 
                ? <span data-testid='info'><b>No users found with this search.</b></span>
                : users.map(user =>
                    <User user={user} key={user.id} page={page} />
            ))}
        </Container>
    )
}

export default UsersList