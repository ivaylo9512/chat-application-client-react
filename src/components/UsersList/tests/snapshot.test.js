import * as redux from 'react-redux';
import { shallow } from 'enzyme';
import UsersList from 'components/UsersList/UsersList';

describe('UsersList snapshot tests', () => {
    const createWrapper = (users, page) => {
        const selectorSpy = jest.spyOn(redux, 'useSelector');
        selectorSpy.mockReturnValueOnce(users);
        selectorSpy.mockReturnValueOnce(page);

        return shallow(<UsersList />)
    }

    it('should render with users', () => {
        const wrapper = createWrapper([{id: 1, firstName: 'test'}, {id: 2, firstName: 'test'}, {id: 3, firstName: 'test'}, {id: 4, firstName: 'test'}, {id: 5, firstName: 'test'}], 1)

        expect(wrapper).toMatchSnapshot();
    })

    it('should render info message when empty users array', () => {
        const wrapper = createWrapper([])
    
        expect(wrapper).toMatchSnapshot();
    })

    it('should not render container with undefined state', () => {
        const wrapper = createWrapper(undefined);

        expect(wrapper).toMatchSnapshot();
    })
})