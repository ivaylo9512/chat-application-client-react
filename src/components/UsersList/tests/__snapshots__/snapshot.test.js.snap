// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList snapshot tests should not render container with undefined state 1`] = `<styled.div />`;

exports[`UsersList snapshot tests should render info message when empty users array 1`] = `
<styled.div>
  <span
    data-testid="info"
  >
    <b>
      No users found with this search.
    </b>
  </span>
</styled.div>
`;

exports[`UsersList snapshot tests should render with users 1`] = `
<styled.div>
  <Memo()
    key="1"
    user={
      Object {
        "firstName": "test",
        "id": 1,
      }
    }
  />
  <Memo()
    key="2"
    user={
      Object {
        "firstName": "test",
        "id": 2,
      }
    }
  />
  <Memo()
    key="3"
    user={
      Object {
        "firstName": "test",
        "id": 3,
      }
    }
  />
  <Memo()
    key="4"
    user={
      Object {
        "firstName": "test",
        "id": 4,
      }
    }
  />
  <Memo()
    key="5"
    user={
      Object {
        "firstName": "test",
        "id": 5,
      }
    }
  />
</styled.div>
`;
